eng/Generate-StatusReport.ps1

```diff
@@ -0,0 +1,173 @@
+    [string]
+    $Repository,
+
+    [Parameter(Mandatory=$false)]
+    [DateTimeOffset]
+    $StartTime,
+
+    [Parameter(Mandatory=$false)]
+    [DateTimeOffset]
+    $EndTime,
+
+    [Parameter(Mandatory=$false)]
+    [string]
+    $GraphQlEndpoint = "https://api.github.com/graphql",
+
+    [Parameter(Mandatory=$false)]
+    [string]
+    $OutputFile,
+
+    [Parameter(Mandatory=$false)]
+    [switch]
+    $Clip
+)
+
+# Author Associations that will not be called out as "external contributions"
+# https://developer.github.com/v4/enum/commentauthorassociation/
+$ExcludedAssociations = @("COLLABORATOR", "OWNER", "MEMBER")
+
+# Individual Users that will not be considered "external contributors"
+$ExcludedAuthors = @("dotnet-maestro")
+
+if (!$EndTime) {
+    $EndTime = New-Object DateTimeOffset ([DateTime]::UtcNow.Date, [TimeSpan]::Zero)
+}
+
+if (!$StartTime) {
+    $StartTime = $EndTime.AddDays(-14)
+}
+
+if (!$Repository) {
+    try {
+        $RemoteUrl = git remote get-url origin
+    } catch {
+        throw "Failed to detect Git Repository. Make sure you're running this from within a clone of a GitHub repository."
+    }
+
+    if ($RemoteUrl -match "^(https?://|ssh://)?(git@)?(www\.)?github.com/(?<owner>[^/]+)/(?<name>[^/]+)$") {
+        $Repository = "$($matches["owner"])/$($matches["name"])"
+    }
+    else {
+        throw "Could not detect GitHub repo identity from remote URL: $RemoteUrl"
+    }
+}
+
+Write-Host -ForegroundColor Green "Generating status report..."
+Write-Host -ForegroundColor Magenta "Start Time: $($StartTime.UtcDateTime) UTC ($($StartTime.ToLocalTime()) Local Time)"
+Write-Host -ForegroundColor Magenta "End Time: $($EndTime.UtcDateTime) UTC ($($EndTime.ToLocalTime()) Local Time)"
+Write-Host -ForegroundColor Magenta "Repository: $Repository"
+
+$SearchQuery = "is:pr is:merged merged:$($StartTime.ToString("yyyy-MM-ddTHH:mm:ssZ"))..$($EndTime.ToString("yyyy-MM-ddTHH:mm:ssZ")) repo:$Repository";
+
+$GraphQlQuery = @"
+query(`$Query: String!) { 
+  search(type:ISSUE, query:`$Query, first: 100) {
+    pageInfo{
+      hasNextPage,
+      endCursor
+    },
+    issueCount,
+    nodes {
+      ... on PullRequest {
+        author{ login, avatarUrl }
+        authorAssociation,
+        number,
+        title,
+        url
+      }
+    }
+  }
+}
+"@
+
+$Parameters = @{
+    "Query" = $SearchQuery
+}
+
+Write-Debug "Executing GraphQL Query:"
+Write-Debug $GraphQlQuery
+Write-Debug "Parameters:"
+$Parameters.Keys | ForEach-Object { Write-Debug "* $_ = $($Parameters[$_])" }
+
+$RequestBody = @{
+    "query" = $GraphQlQuery;
+    "variables" = $Parameters;
+} | ConvertTo-Json
+$Headers = @{
+    "Authorization" = "Bearer $GitHubToken";
+    "Content-Type" = "application/json";
+}
+
+$Response = Invoke-RestMethod -Method "POST" -Headers $Headers -Body $RequestBody $GraphQlEndpoint
+
+$Cursor = $null;
+$ContributorPRs = @();
+do {
+    if ($Cursor) {
+        throw "Multi-page results not yet supported!"
```

Comment:
FYI, I just got lazy here and figured I wouldn't be running the script over a period that contained 100 PRs. It's possible it could happen and we'd need to do paging. It's [not terribly hard though](https://graphql.github.io/learn/pagination/)